networking/hardware_networks/uninstalling-sriov-operator.adoc

```diff
@@ -0,0 +1,43 @@
+[id="uninstalling-sriov-operator"]
+= Uninstalling the SR-IOV Network Operator
+include::modules/common-attributes.adoc[]
+:context: uninstalling-sr-iov-operator
+
+toc::[]
+
+To uninstall the SR-IOV Network Operator, you must first delete any running SR-IOV workloads. You must also delete the webhooks after uninstalling the operator.
+
+.Prerequisites
+
+* Access to an {product-title} cluster using an account with `cluster-admin` permissions.
+* The SR-IOV Network Operator Operator must be installed.
+
+.Procedure
+
+. Delete all SR-IOV custom resources (CRs).
```

Comment:
Due to [BZ2039950](https://bugzilla.redhat.com/2039950) we might have an issue with removing the CRs.
If the operator is installed from a subscription, the finalizers on the SriovNetwork might not be removed, and the deletion will hang on the remaining finalizer in SriovNetwork. Should we add this to the description?

Reply:
What concretely would you add? What would be the steps/commands?

Reply:
@mmirecki gentle reminder :-)

Reply:
Sorry for the delay. Below is a suggestion on the note:

Note: Due to BZ2039950 the sriov operator might not perform a full cleanup after being removed. Some SriovNetworks might be stuck  pending deletetion due to stray finalizers. To remove these SriovNetworks please execute the following for each of the remaining networks:
`oc patch SriovNetwork -n <namespace> <network name>  -p '{"metadata":{"finalizers":null}}' --type=merge`

Reply:
An automated way to patch all networks would be:
`oc get SriovNetwork -A -o name | sed 's:.*/::' | xargs oc patch SriovNetwork -n openshift-sriov-network-operator  -p '{"metadata":{"finalizers":null}}' --type=merge`

Reply:
An alternative way to cleanup, replacing both this and the webhook changes would be to scale down the deployment and let it cleanup up:
oc scale deployment --replicas=0 -n openshift-sriov-network-operator                   sriov-network-operator